test/e2e/framework/nodes_util.go

```diff
@@ -119,17 +119,34 @@ func masterUpgradeGCE(rawV string, enableKubeProxyDaemonSet bool) error {
 	return err
 }
 
+func locationParamGKE() string {
+	if TestContext.CloudConfig.Zone != "" {
+		return fmt.Sprintf("--zone=%s", TestContext.CloudConfig.Zone)
+	}
+	return fmt.Sprintf("--region=%s", TestContext.CloudConfig.Region)
+}
+
+func gcloudContainerCommandGroup() string {
+	if TestContext.CloudConfig.Region != "" {
+		// TODO(wojtek-t): Get rid of it once Regional Clusters go to GA.
+		return "beta"
+	}
+	return ""
+}
+
 func masterUpgradeGKE(v string) error {
 	Logf("Upgrading master to %q", v)
-	_, _, err := RunCmd("gcloud", "container",
+	args := []string{
+		"container",
 		"clusters",
 		fmt.Sprintf("--project=%s", TestContext.CloudConfig.ProjectID),
-		fmt.Sprintf("--zone=%s", TestContext.CloudConfig.Zone),
+		locationParamGKE(),
 		"upgrade",
 		TestContext.CloudConfig.Cluster,
 		"--master",
 		fmt.Sprintf("--cluster-version=%s", v),
 		"--quiet")
+	_, _, err := RunCmd("gcloud", gcloudContainerCommandGroup(), args...)
```

Comment:
Does it work for zonal case?

When I'm calling "gcloud '' container clusters list" I'm getting: ERROR: (gcloud) Invalid choice: ''.

Reply:
It should already be fixed now :)